### PR TITLE
[FIX] send events after completion of waiting time

### PR DESCRIPTION
**Pull request by sindhu-106**
Signed-off-by: sindhuja <sindhuja.gajula@intel.com>

### CODE REVIEW

**Review by paul-amonson — DISMISSED** [review]
Looks good.

**Review by paul-amonson — DISMISSED** [review]

File: eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java [hunk 1]
```diff
@@ -158,13 +158,16 @@ private void publishEventsForScenario(Map<String, String> parameters, String sce
                                           PropertyMap events, PropertyMap scenarioParameters)
             throws PropertyNotExpectedType, SimulatorException {
         PropertyArray publishEvents;
+        String startTime =  parameters.getOrDefault("start-time", scenarioParameters.getString("start-time"));
```
Comment (nbwhite) [inline eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java:161]: This is a new instance of getOrDefault, so don't forget to update the ipdlt exclusions in gus/dai. Let me or Paul know if you need help doing that.

File: eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java [hunk 2]
```diff
@@ -280,45 +283,40 @@ private void generateEventsForRepeatMode(PropertyMap events, Map<String, String>
...
+        PropertyArray events_ = new PropertyArray();
         if(repeatModeScenario.equals("burst"))
-            publishEvents = foreignScenario_.generateEventsForBurstMode(events, repeatModeScenario);
+            events_ = foreignScenario_.generateEventsForBurstMode(events, repeatModeScenario);
 
         else if (repeatModeScenario.equals("group-burst"))
-            publishEvents = foreignScenario_.generateEventsForGroupBurstMode(events, repeatModeScenario);
+            events_ = foreignScenario_.generateEventsForGroupBurstMode(events, repeatModeScenario);
 
+        String startTime = parameters.getOrDefault("start-time", scenarioParameters.getString("start-time"));
```
Comment (nbwhite) [inline eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java:293]: Another getOrDefault instance

File: eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java [hunk 3]
```diff
@@ -280,45 +283,40 @@ private void generateEventsForRepeatMode(PropertyMap events, Map<String, String>
...
-
-        if(clockModeStartTime != null)
-            clockMode = "start-time";
-
-        if (clockModeCounter != null)
+        if (parameters.getOrDefault("counter", null) != null)
             clockMode = "counter";
 
-        if(clockModeDuration != null)
+        if(parameters.getOrDefault("duration", null) != null)
```
Comment (nbwhite) [inline eventsim/src/main/java/com/intel/dai/eventsim/ForeignSimulatorEngine.java:300]: Another one

**Review by nbwhite — DISMISSED** [review]

**Review by paul-amonson — APPROVED** [review]

**Review by nbwhite — APPROVED** [review]

**Pull request merged**